scripts/databricks_install.py

```diff
@@ -0,0 +1,255 @@
+      default="Recommenders.egg")
+  parser.add_argument("--dbfs-path",
+      help="The directory on dbfs that want to place files in",
+      default="dbfs:/FileStore/jars")
+  parser.add_argument("--overwrite",
+      action="store_true",
+      help="Whether to overwrite existing files.",
+      default=False)
+  parser.add_argument("--prepare-o16n",
+      action="store_true",
+      help="Whether to install additional libraries for operationalization.",
+      default=False)
+  parser.add_argument("cluster_id", 
+      help="cluster id for the cluster to install on.")
+  args = parser.parse_args()
+
+  ###############################
+  ## Create the egg:
+  ###############################
+
+  print("Preparing Recommenders library file ({})...".format(args.eggname))
+  myegg = create_egg(args.path_to_recommenders, local_eggname=args.eggname, overwrite=args.overwrite)
+  print('Created: {}'.format(myegg))
+
+  ###############################
+  ## Interact with Databricks:
+  ###############################
+
+  ## first make sure you are using the correct profile and connecting to the intended workspace
+  my_api_client = _get_api_client(ProfileConfigProvider(args.profile).get_config())
+
+  ## upload the egg:
+  uploadpath='/'.join([args.dbfs_path, os.path.basename(myegg)])
+  print("Uploading {} to databricks at {}".format(os.path.basename(myegg), uploadpath))
+  DbfsApi(my_api_client).cp(recursive=False, src=myegg, dst=uploadpath, overwrite=args.overwrite)
+
+  ## steps below require the cluster to be running. Check status
+  try:
+    status=ClusterApi(my_api_client).get_cluster(args.cluster_id)
+  except HTTPError as e:
+    print(e)
+    print(textwrap.dedent(CLUSTER_NOT_FOUND_MSG.format(args.cluster_id)))
+    raise
+    
+  if status['state'] != 'RUNNING':
+    print(textwrap.dedent(CLUSTER_NOT_RUNNING_MSG.format(args.cluster_id, status['state'])))
+    sys.exit()
+    
+  ## install the library:
+  print("Installing the reco_utils module onto databricks cluster {}".format(args.cluster_id))
+  formatted_lib = {'egg': uploadpath}
+  LibrariesApi(my_api_client).install_libraries(args.cluster_id, [formatted_lib])
+  
+  ## prepare for operationalization if desired:
+  if args.prepare_o16n:
+    prepare_for_operationalization(
+      cluster_id=args.cluster_id, api_client=my_api_client,
+      status=status, dbfs_path=args.dbfs_path, overwrite=args.overwrite)
+
+  ## restart the cluster for new installation(s) to take effect.
+  print("Restarting databricks cluster {}".format(args.cluster_id))
+  ClusterApi(my_api_client).restart_cluster(args.cluster_id)
+
+  ## wrap up and send out a final message:
+  print("""
+  Requests submitted. You can check on status of your cluster with: 
+
+  databricks --profile """+args.profile+""" clusters list
+  """
+  )
+    
+    
+sys.exit()
+
+#######################################################
+## STOP HERE
```

Comment:
are these comments needed?

Reply:
no, i was just keeping them for adding a feature in the future to actually create a cluster. I will add that and clean this up. Just wanted feedback first

Reply:
I mean, not in the future but now ;-)

Reply:
done for an initial attempt. Right now, it's not configurable, and it creates a cluster with the same default values they would get in the portal. We can add customization later. The biggest issue with configuring customizable clusters is that there are so many variables that it becomes tedious to parse them all. I think we'd want people to pass a `--cluster-config xx.json` argument that provides a path to a json file that can be parsed. Does that seem reasonable @loomlike?